src/services/users.js

```diff
@@ -0,0 +1,191 @@
+      } else {
+        res.boom.badRequest('user is not active');
+      }
+    } catch (err) {
+      return res.boom.notFound();
+    }
+  }
+
+  async updateUserPassword (req, res) {
+    try {
+      const currentStatus = await this.getCurrentStatus(req);
+      const statusActive = await this.getStatus('Active');
+      if (currentStatus.toString() === statusActive._id.toString()) {
+        const { oldPassword, newPassword } = req.body;
+        const user = await User.findOne({ _id: req.params.id }).exec();
+        const isMatchPasswords = await bcrypt.compare(
+          oldPassword,
+          user.password
+        );
+        if (!isMatchPasswords) {
+          return res.boom.badRequest('wrong old password');
+        }
+        if (isMatchPasswords && oldPassword !== newPassword) {
+          user.password = newPassword;
+          const salt = await bcrypt.genSalt(10);
+          user.password = await bcrypt.hash(user.password, salt);
+          const updatedUserPassword = User.findOneAndUpdate(
+            { _id: req.params.id },
+            user
+          );
+          return updatedUserPassword ? res.json(true) : res.boom.notFound();
+        } else {
+          res.boom.badRequest('new password cannot be the same as old');
+        }
+      }
+    } catch {
+      res.boom.notFound();
+    }
+  }
+
+  async blockUser (req, res) {
+    try {
+      const statusBlocked = await this.getStatus('Blocked');
+      const currentStatus = await this.getCurrentStatus(req);
+      if (statusBlocked._id.toString() !== currentStatus.toString()) {
+        const user = await User.findByIdAndUpdate(
+          req.params.id,
+          { status: statusBlocked._id },
+          {
+            new: true
+          }
+        ).exec();
+        return user ? res.json(true) : res.boom.notFound();
+      } else {
+        return res.boom.badRequest('user already blocked');
+      }
+    } catch (err) {
+      res.boom.notFound();
+    }
+  }
+
+  async deleteUser (req, res) {
+    try {
+      const statusDeleted = await this.getStatus('Deleted');
+      const currentStatus = await this.getCurrentStatus(req);
+      if (statusDeleted._id.toString() !== currentStatus.toString()) {
+        const user = await User.findByIdAndUpdate(
+          req.params.id,
+          { status: statusDeleted._id },
+          {
+            new: true
+          }
+        ).exec();
+        return user ? res.json(true) : res.boom.notFound();
+      } else {
+        return res.boom.badRequest('user already deleted');
+      }
+    } catch (err) {
+      res.boom.notFound();
+    }
+  }
+
+  async createBasicUser (req, res) {
+    const statusActive = await this.getStatus('Active');
+    const newUser = new User(req.body);
+    newUser.status = statusActive._id;
+    const salt = await bcrypt.genSalt(10);
+    newUser.password = await bcrypt.hash(newUser.password, salt);
+    return newUser;
+  }
+
+  async checkUserExists (req, res) {
```

Comment:
I don't think that we need these 4 methods (checkUserExists, getStatus, getRole, getCurrentStatus). They don't make code shorter

Reply:
Fixed